lua/telescope/builtin/init.lua

```diff
@@ -936,27 +936,18 @@ end
 
 builtin.tags = function(opts)
   opts = opts or {}
-
   local ctags_file = opts.ctags_file or 'tags'
-
-  if not vim.loop.fs_open(vim.fn.expand(ctags_file), "r", 438) then
+  opts.entry_maker = opts.entry_maker or make_entry.gen_from_ctags(opts)
+  if vim.loop.fs_stat(vim.fn.expand(ctags_file, true)) == nil then
     print('Tags file does not exists. Create one with ctags -R')
     return
   end
-
-  local fd = assert(vim.loop.fs_open(vim.fn.expand(ctags_file), "r", 438))
-  local stat = assert(vim.loop.fs_fstat(fd))
-  local data = assert(vim.loop.fs_read(fd, stat.size, 0))
-  assert(vim.loop.fs_close(fd))
-
-  local results = vim.split(data, '\n')
-
   pickers.new(opts,{
     prompt = 'Tags',
-    finder = finders.new_table {
-      results = results,
-      entry_maker = make_entry.gen_from_ctags(opts),
-    },
+    finder = finders.new_oneshot_job(
+      { 'cat', ctags_file },
```

Comment:
We don't expand here maybe we should do `local ctags_file = vim.fn.expand(opts.ctags_file, true) or 'tags'` in line 940. Then we could just have on expand.

Reply:
Done

Reply:
@tami5 won't work. because `fn.expand` will return null rather than nil which does not work with or. Could you please figure out another way :smile: thanks

Reply:
Oh you mean for the lua loop issue?

Reply:
@Conni2461

Reply:
Also @Conni2461 I noticed a lot of repo locate their tag file in doc/tag, i think we should add the additional step of checking ctags_file again

Reply:
We cannot do this because if opts.ctags_file is not specified `expand(...)` will return `null` and not `nil` so we never get into the second condition. Try it out

Reply:
Updated ` opts.ctags_file and opts.ctags_file:gsub("~", vim.loop.os_homedir()) or 'tags'`